lib/validator/dbs/block_hashes.rs

```diff
@@ -280,6 +291,60 @@ impl BlockHashDbs {
         })
     }
 
+    /// Find the last common ancestor of two blocks,
+    /// if headers for both exist
+    pub fn last_common_ancestor(
+        &self,
+        rotxn: &RoTxn,
+        block_hash0: BlockHash,
+        block_hash1: BlockHash,
+    ) -> Result<BlockHash, error::LastCommonAncestor> {
+        use std::cmp::Ordering;
+        // if the block hashes are the same, return early
+        if block_hash0 == block_hash1 {
+            return Ok(block_hash0);
+        }
+        if block_hash0 == BlockHash::all_zeros() || block_hash1 == BlockHash::all_zeros() {
+            return Ok(BlockHash::all_zeros());
+        }
+        let height0 = self.height.get(rotxn, &block_hash0)?;
+        let height1 = self.height.get(rotxn, &block_hash1)?;
+        let mut ancestors0 = self.ancestor_headers(rotxn, block_hash0);
+        let mut ancestors1 = self.ancestor_headers(rotxn, block_hash1);
+        // Equalize heights to min(height0, height1)
```

Comment:
This comment looks like it should be placed further down? What you're doing here appears to be verifying that both sets of ancestors have enough items?

Reply:
Okay, scratch that. What you're doing is iterating on the tallest block hash ancestor set, in a way that they actually end up equalized. Not really immediately clear, a longer comment would be great